cobra/exceptions.py

```diff
@@ -32,6 +32,12 @@ class SolverNotFound(Exception):
     pass
 
 
+class BoundsError(Exception):
```

Comment:
I'm not a huge fan of introducing too specific error classes. Maybe we need a base `COBRAError` class and a more descriptive message per error. What do others think? @opencobra/cobrapy-core

Reply:
I dont't see any problem with having a base class for cobrapy related errors. Then, we could roughly divide the errors into errors from solver optimization and other basic config errors.

Reply:
Aren't there solver specific errors already? There is an OptimizationError I think. If any we might want to have a FeasibilityError which includes bounds as well as equalities, but I think a generic SolverError or SolutionError would also work.

Reply:
+1 for specific error classes that also inherit from a cobrapy base error. It could also make sense that this specific error inherits from `ValueError` as well.
I would add that the exception thrown here should be consistent with the [similar check when setting reaction bounds](https://github.com/opencobra/cobrapy/blob/4481472c90ae653f92aacc6bdb4210dbb715e849/cobra/core/reaction.py#L190).